src/test/java/teammates/test/pageobjects/InstructorCourseStudentDetailsEditPage.java

```diff
@@ -73,10 +73,19 @@ private void fillStudentDetailsForm(String studentName, String teamName, String
             fillTextBox(commentsTextbox, comments);
         }
         if (teamName == null) {
-            click(submitButton);
+            if (studentEmail == null) {
+                click(submitButton);
+            } else {
+                clickAndConfirm(submitButton);
+            }
+            
         } else {
             // if team name is edited, the confirmation dialog will pop up
-            clickAndConfirm(submitButton);
+            if (studentEmail == null) {
+                clickAndConfirm(submitButton);
+            } else {
+                clickAndConfirmTwoTimes(submitButton);
```

Comment:
`waitForModalToDisappear()` in `waitForConfirmationModalAndClickOk()` is causing the issue there. As whenever first modal disappears the second modal appear, that time `waitForModalToDisappear()` keep on waiting to disappear which will not gonna happen.